logstash-core/src/main/java/org/logstash/config/ir/graph/IfVertex.java

```diff
@@ -20,6 +21,9 @@ public BooleanExpression getBooleanExpression() {
 
     public IfVertex(SourceMetadata meta, BooleanExpression booleanExpression) {
         super(meta);
+        this.id = "if-" +
+                (meta != null ? Integer.toString(meta.hashCode()) : UUID.randomUUID().toString())
```

Comment:
I think this implementation is OK for now.

The only down side I currently see the implementation is tied to the ruby language implementation.
Maybe we should hide that generation logic behind a method that we can call on both BooleanExpression and IfVertex this will give us more future flexibility? WDYT?